### ISSUE TITLE
Cron Job group-sync Resource Quota Error in rhods-notebooks Namespace

### ISSUE THREAD

**Issue by schwesig** [nerc-project/operations#467]
I am not sure how urgent that is, but I want to bring it from the [slack discussion](https://massopencloud.slack.com/archives/C06EFPN5G2D/p1709831289784469) into this issue:
# Cron Job `group-sync` Resource Quota Error in `rhods-notebooks` Namespace

## Summary
The `group-sync` cron job within the `rhods-notebooks` namespace is encountering quota-related errors when attempting to create pods. The `Jgroup-sync-28496700` has been occurring frequently lately, with 153 instances reported in the last 9 hours as of March 7, 2024, at 1:46 PM. The error message indicates a failure to specify `limits.cpu` and `limits.memory` for the `group-sync` pods, resulting in a forbidden operation due to quota restrictions.

![image](https://github.com/nerc-project/operations/assets/89909507/d0574523-2694-4d54-9b91-e3dcbb1d99d4)
https://console.apps.shift.nerc.mghpcc.org/k8s/ns/rhods-notebooks/jobs/group-sync-28496700/events

## Details
- **Error Message**: `Error creating: pods "group-sync-28496700-twdsc" is forbidden: failed quota: ope-course-quota: must specify limits.cpu, limits.memory for: group-sync`
- **Frequency**: The error has been generated 153 times in the last 9 hours.
- **Cron Job Schedule**: Runs every hour.
- **Observation Start**: The issue started occurring recently, within the last 1 to 3 days on different pods. (to be checked for more details)
- **Initial Deployment**: The cron job was created on January 24, 2024, and has been running without this issue from what I see.

## Possible Cause and Solution

From Danni: Not sure why the cronjob was not giving this error before. I think adding the resource limits in cronjob.yaml could solve this problem: 

```yaml
imagePullPolicy: Always 
resources:
  limits:
    memory: "800Mi"
  requests:
    memory: "400Mi"
```

## Action Items
1. **Update Cron Job Configuration**: Implement the proposed changes in the `cronjob.yaml` to include `limits.memory` specifications.
2. **Monitor for Resolution**: After applying the updates, monitor the cron job executions to ensure the error is resolved and no further quota-related issues arise.

**Comment by DanNiESh** [comment]
Oh, I found that this cronjob in rhods-notebook namespace is actually deprecated. Since there are group-sync cronjobs under each class namespace. See https://console.apps.shift.nerc.mghpcc.org/k8s/ns/cs210-computer-system-boston-university-e5839d/cronjobs,  https://console.apps.shift.nerc.mghpcc.org/k8s/ns/ece440spring2024-619f12/cronjobs/group-sync/, https://console.apps.shift.nerc.mghpcc.org/k8s/ns/cs506-dc03b3/cronjobs. I think we can simply delete this cronjob under rhods-notebooks ns.

**Comment by dystewart** [comment]
@schwesig @DanNiESh Yeah that's correct, just verified all the cronjobs are still working in the course namespaces. I will delete the job in the rhods-notebooks namespace.

**Comment by dystewart** [comment]
Deleted, closing this issue
